### PR TITLE
[REG master] Issue14817 std.algorithm.copy/std.file.copy conflict

### PR DESCRIPTION
**Pull request by DmitryOlshansky**
Reboot of #3509 now with a unittest.

### CODE REVIEW

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled off

**Comment by quickfur** [comment]
LGTM

**Comment by DmitryOlshansky** [comment]
Merge anyone?

File: std/file.d [hunk 1]
```diff
@@ -2887,6 +2887,7 @@ private void copyImpl(const(char)[] f, const(char)[] t, const(FSChar)* fromz, co
 
 unittest
 {
+    import std.algorithm, std.file; // issue 14817
```
Comment (schveiguy) [inline std/file.d]: I can't believe you have to import std.file inside std.file. That seems... wrong.

Comment (DmitryOlshansky) [inline std/file.d]: Compiler bugs in local imports - std.algo becomes a bunch of local aliases sooo it shadows current top-level symbols.

**Comment by schveiguy** [comment]
Auto-merge toggled on

File: std/file.d (hunk 1 above)
Comment (schveiguy) [inline std/file.d]: Oh, I know. It's just unbelievable :)

**Pull request merged**